service/src/intTest/java/uk/nhs/adaptors/gpc/consumer/gpc/GpcConfigurationValidationTest.java

```diff
@@ -0,0 +1,259 @@
+            .run(context -> {
+                assertThat(context).hasFailed();
+
+                var startupFailure = context.getStartupFailure();
+
+                assertThat(startupFailure)
+                    .rootCause()
+                    .hasMessageContaining(
+                        "Either all or none of the GPC_CONSUMER_SPINE_ variables must be defined. "
+                            + "Missing variables: GPC_CONSUMER_SPINE_CLIENT_CERT"
+                    );
+            });
+    }
+
+    @Test
+    void When_GpcConfigurationHasAnInvalidCertificate_Expect_ContextIsNotCreated() {
+        contextRunner
+            .withPropertyValues(
+                buildPropertyValue("clientCert", INVALID_CERTIFICATE),
+                buildPropertyValue("clientKey", VALID_RSA_PRIVATE_KEY),
+                buildPropertyValue("rootCA", VALID_CERTIFICATE),
+                buildPropertyValue("subCA", VALID_CERTIFICATE)
+            )
+            .run(context -> {
+                assertThat(context).hasFailed();
+
+                var startupFailure = context.getStartupFailure();
+
+                assertThat(startupFailure)
+                    .rootCause()
+                    .hasMessageContaining(
+                        "One or more of the GPC_CONSUMER_SPINE_ variables is in an invalid PEM format. "
+                            + "Invalid variables: GPC_CONSUMER_SPINE_CLIENT_CERT"
+                    );
+            });
+    }
+
+    @Test
+    void When_GpcConfigurationHasAnInvalidClientKey_Expect_ContextIsNotCreated() {
+        contextRunner
+            .withPropertyValues(
+                buildPropertyValue("clientCert", VALID_CERTIFICATE),
+                buildPropertyValue("clientKey", INVALID_RSA_PRIVATE_KEY),
+                buildPropertyValue("rootCA", VALID_CERTIFICATE),
+                buildPropertyValue("subCA", VALID_CERTIFICATE)
+            )
+            .run(context -> {
+                assertThat(context).hasFailed();
+
+                var startupFailure = context.getStartupFailure();
+
+                assertThat(startupFailure)
+                    .rootCause()
+                    .hasMessageContaining(
+                        "One or more of the GPC_CONSUMER_SPINE_ variables is in an invalid PEM format. "
+                            + "Invalid variables: GPC_CONSUMER_SPINE_CLIENT_KEY"
+                    );
+            });
+    }
+
+    @Test
+    void When_GpcConfigurationHasSspUrlPresentWithTrailingSlash_Expect_ContextIsCreatedAndIsSspEnabled() {
+        contextRunner
+            .withPropertyValues(
+                buildPropertyValue("sspUrl", "/this-is-a-url.com/")
+            )
+            .run(context -> {
+                assertThat(context)
+                    .hasNotFailed()
+                    .hasSingleBean(GpcConfiguration.class);
+
+                var gpcConfiguration = context.getBean(GpcConfiguration.class);
+                assertThat(gpcConfiguration.isSspEnabled()).isTrue();
```

Comment:
I think this can be replaced with a simple "assertTrue" which is a bit more concise version